src/platform/plugins/shared/unified_search/public/query_string_input/query_bar_top_row.tsx

```diff
@@ -232,6 +232,10 @@ export interface QueryBarTopRowProps<QT extends Query | AggregateQuery = Query>
      */
     controlsWrapper: React.ReactNode;
   };
+  /**
+   * Optional - Callback function invoked to open the current query in a new Discover tab
+   */
+  onOpenQueryInNewTab?: ESQLEditorProps['onOpenQueryInNewTab'];
```

Comment:
All the esql related props are mentioning esql, so it would be better to also add the esql word in this property/